xlml/utils/xpk.py

```diff
@@ -43,15 +43,9 @@ def run_workload(
     task_owner: str,
     startup_timeout: int,
     num_slices: int = 1,
-) -> KubernetesPodOperator:
-  """Run workload through xpk tool.
-
-  The reason to use KubernetesPodOperator instead of BashOperator is that
-  xpk must run with Python 3.10 or greater; however, the latest version in
-  Composer is Python 3.8, and it's non-trivial to upgrade it as the  Composer
-  uses docker images that bundle Airflow releases with Python and other
-  libraries.
-  """
+):
+  """Run workload through xpk tool."""
+  import subprocess
```

Comment:
Can subprocess be imported at the global scope? If not, can you comment why?

Reply:
Ah spot on! I'll add a comment. The `task.docker/kubernetes` decorators don't capture any scope beyond the method they decorate, so the imports must be local.

Reply:
I think it's best practice for Airflow to import in a smaller scope for expensive dependency. We could do that as well in refactoring. 

```
Note that import statements also count as top-level code. So, if you have an import statement that takes a long time or the imported module itself executes code at the top-level, that can also impact the performance of the scheduler. The following example illustrates how to handle expensive imports.
```
from [here](https://airflow.apache.org/docs/apache-airflow/stable/best-practices.html)

Reply:
Interesting! From what I can tell, the `task` bodies aren't evaluated until DAG execution time, so it makes sense that imports should be local.